domain/data-access/form-repository.ts

```diff
@@ -92,38 +91,50 @@ const updateFormInstance = async (
   await query(deltaQuery);
 };
 
-const addFormInstance = async (
-  validatedContent: string,
-  instanceUri: string,
-  formLabel: string,
-) => {
-  const predicate = 'http://mu.semte.ch/vocabularies/ext/label';
-  const updatedContent = addTripleToTtl(
-    validatedContent,
-    instanceUri,
-    predicate,
-    formLabel,
-  );
-
-  await query(ttlToInsert(updatedContent));
+const addFormInstance = async (instanceContent: string) => {
+  await query(ttlToInsert(instanceContent));
 };
 
 const deleteFormInstance = async (formTtl: string, instanceUri: string) => {
   const q = await buildFormDeleteQuery(formTtl, instanceUri);
   await query(q);
 };
 
-const getFormInstances = async (formLabel: string) => {
+const getFormInstanceCount = async (
+  targetType: string,
+  labelPredicate: string,
+) => {
   const q = `
     PREFIX inst: <http://data.lblod.info/form-data/instances/>
     PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
     PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    SELECT (COUNT(DISTINCT ?uri) as ?count)
+    WHERE {
+        ?uri a ${sparqlEscapeUri(targetType)} .
+        ?uri ${sparqlEscapeUri(labelPredicate)} ?label .
+        ?uri mu:uuid ?id .
+    }`;
+
+  const queryResult = await query(q);
+
+  return parseInt(queryResult.results.bindings[0]?.count?.value || '0');
```

Comment:
It would give more clarity if this line becomes 
`return parseInt(queryResult.results.bindings[0]?.count?.value) || 0`